src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/UpdateOrganisationTest.java

```diff
@@ -0,0 +1,228 @@
+import uk.gov.hmcts.reform.professionalapi.persistence.PaymentAccountRepository;
+import uk.gov.hmcts.reform.professionalapi.persistence.ProfessionalUserRepository;
+import uk.gov.hmcts.reform.professionalapi.util.ProfessionalReferenceDataClient;
+import uk.gov.hmcts.reform.professionalapi.util.Service2ServiceEnabledIntegrationTest;
+
+public class UpdateOrganisationTest extends Service2ServiceEnabledIntegrationTest {
+
+    @Autowired
+    private OrganisationRepository organisationRepository;
+
+    @Autowired
+    private ProfessionalUserRepository professionalUserRepository;
+
+    @Autowired
+    private ContactInformationRepository contactInformationRepository;
+
+    @Autowired
+    private DxAddressRepository dxAddressRepository;
+
+    @Autowired
+    private PaymentAccountRepository paymentAccountRepository;
+
+    private ProfessionalReferenceDataClient professionalReferenceDataClient;
+
+    @Before
+    public void setUp() {
+        professionalReferenceDataClient = new ProfessionalReferenceDataClient(port);
+        dxAddressRepository.deleteAll();
+        contactInformationRepository.deleteAll();
+        professionalUserRepository.deleteAll();
+        paymentAccountRepository.deleteAll();
+        organisationRepository.deleteAll();
+    }
+
+    @Test
+    public void updates_non_existing_organisation_returns_status_404() {
+        updateAndValidateOrganisation(UUID.randomUUID().toString(),OrganisationStatus.ACTIVE,404);
+    }
+
+    @Test
+    public void updates_organisation_with_organisation_identifier_null_returns_status_400() {
+        updateAndValidateOrganisation(null,OrganisationStatus.ACTIVE,400);
+    }
+
+    @Test
+    public void updates_organisation_with_invalid_organisation_identifier_returns_status_400() {
+        updateAndValidateOrganisation("1234ab12",OrganisationStatus.ACTIVE,400);
+    }
+
+    @Test
+    public void can_update_organisation_status_from_pending_to_active_should_returns_status_200() {
+        updateAndValidateOrganisation(createOrganisationRequest(),OrganisationStatus.ACTIVE,200);
+    }
+
+    @Test
+    public void can_update_organisation_status_from_active_to_blocked_should_returns_status_200() {
+        String organisationIdentifier = createOrganisationRequest();
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.ACTIVE,200);
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.BLOCKED,200);
+    }
+
+    @Test
+    public void can_update_organisation_status_from_active_to_deleted_should_returns_status_200() {
+        String organisationIdentifier = createOrganisationRequest();
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.ACTIVE,200);
+        updateAndValidateOrganisation(organisationIdentifier,OrganisationStatus.DELETED,200);
+    }
+
+    @Test
+    public void can_update_organisation_status_from_pending_to_pending_should_returns_status_200() {

```

Comment:
A Test for Pending to Deleted is missing according to the requirements.

Reply:
Added